pandas/io/parsers.py

```diff
@@ -1474,6 +1474,13 @@ def _set(x):
                 else:
                     _set(val)
 
+        elif self.parse_dates:
+            if isinstance(self.index_col, list):
```

Comment:
We expect `parse_dates` to a be `bool`, `list`, or `dict` per documentation.  This is explicitly validated as well, so `is_list_like` is unnecessary.